src/test/java/core/basesyntax/service/RegistrationServiceImplTest.java

```diff
@@ -0,0 +1,262 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.RegistrationException;
+import core.basesyntax.model.User;
+import java.util.ArrayList;
+import java.util.List;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RegistrationServiceImplTest {
+    private static StorageDao storageDao = new StorageDaoImpl();
+    private final RegistrationService registrationService = new RegistrationServiceImpl();
+    private List<User> users = new ArrayList<>();
+
+    @BeforeEach
+    void setUp() {
+        Storage.people.clear();
+
+        User user1 = new User();
+        user1.setAge(17);
+        user1.setLogin("12345");
+        user1.setPassword("12345");
+
+        User user2 = new User();
+        user2.setAge(18);
+        user2.setLogin("123456");
+        user2.setPassword("123456");
+
+        User user3 = new User();
+        user3.setAge(19);
+        user3.setLogin("1234567");
+        user3.setPassword("1234567");
+
+        User user4 = new User();
+        user4.setAge(20);
+        user4.setLogin("12345678");

```

Comment:
Why do you need all these users?
Better create here 1 valid user (do not save it to DB) and use it in the all 'not ok' tests - simply set one invalid field to this user.